### PR TITLE
Feature/add security scan vulnerability calls

### PR DESCRIPTION
**Pull request by tbronsonCoveo**
### Problem
Missing Github advanced security attributes and calls

### Solution
- Added attributes to Repository object to include SecurityAndAnalysis values:
  - advanced security status
  - secret scanning status
  - secret scanning push protection status
- Added _get code scanning analyses_ call to Repository
  - Created CodeScanningAnalysis Object returned from call 

Tests are incoming, opening now to get immediate feedback as I try to understand the custom test framework.

### CODE REVIEW

**Comment by tbronsonCoveo** [comment]
Tests are complete

FYI

- This repo is pretty far from the original fork
- I tried applying the linting following the doc but it wanted to change another 15+ files so I didn't
- The tests are broken locally (for me at least) unless you remove the assertion in Framework.py:176 that validates the header
- I added tests that cover all but didn't bother adding tests to each small object class since they are already tested in Repository
- I couldn't get the replay/record feature to work (does it even exist?) so I just copy pasted template queries from Github doc

**Review by JPLachance — APPROVED** [review]
From what I understand of PyGitHub, this looks good to me.

**Review by jonapich — APPROVED** [review]
💯

File: github/AdvancedSecurity.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import github.Commit
+import github.GithubObject
+
+
+class AdvancedSecurity(github.GithubObject.NonCompletableGithubObject):
+    """
+    This class represents AdvancedSecurity.
```
[the remark this answers is not in the document]
Comment (tbronsonCoveo) [reply github/AdvancedSecurity.py:7]: I tried to find something but it's just a random entry added to the repository return. So not as far as I know.

**Comment by tbronsonCoveo** [comment]
@jonapich Donc est-ce qu'on peut merge? Je connais pas la procédure pour tag une release and deploy au pypi coveo.

**Pull request merged**